src/components/Viewer/InformationPanel/Annotation/Item.tsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import { Item as ItemStyled } from "src/components/Viewer/InformationPanel/Annotation/Item.styled";
+import { ViewerContextStore, useViewerState } from "src/context/viewer-context";
+import {
+  AnnotationNormalized,
+  type CanvasNormalized,
+  EmbeddedResource,
+  InternationalString,
+} from "@iiif/presentation-3";
+import AnnotationItemPlainText from "./PlainText";
+import AnnotationItemHTML from "./HTML";
+import AnnotationItemVTT from "./VTT/VTT";
+import { parseAnnotationTarget } from "src/lib/annotation-helpers";
+import { createOpenSeadragonRect } from "src/lib/openseadragon-helpers";
+import AnnotationItemImage from "./Image";
+
+type Props = {
+  annotation: AnnotationNormalized;
+};
+
+export const AnnotationItem: React.FC<Props> = ({ annotation }) => {
+  const viewerState: ViewerContextStore = useViewerState();
+  const { openSeadragonViewer, vault, activeCanvas, configOptions } =
+    viewerState;
+
+  const annotationBody: Array<
+    EmbeddedResource & {
+      label?: InternationalString;
+    }
+  > = annotation.body.map((body) => vault.get(body.id));
+  const annotationBodyFormat =
+    annotationBody.find((body) => body.format)?.format || "";
+  const annotationBodyValue =
+    annotationBody.find((body) => body.value)?.value || "";
+
+  function renderItemBody(annotation: AnnotationNormalized) {
+    const { target } = annotation;
+
+    const canvas: CanvasNormalized = vault.get({
+      id: activeCanvas,
+      type: "Canvas",
+    });
+
+    function handleClick() {
+      if (!target) return;
+
+      const zoomLevel = configOptions.annotationOverlays?.zoomLevel || 1;
+      // @ts-ignore
+      const parsedAnnotationTarget = parseAnnotationTarget(target);
+
+      const { point, rect, svg } = parsedAnnotationTarget;
+
+      if (point || rect || svg) {
+        const rect = createOpenSeadragonRect(
+          canvas,
+          parsedAnnotationTarget,
+          zoomLevel,
+        );
+        openSeadragonViewer?.viewport.fitBounds(rect);
+      }
+    }
+
+    switch (annotationBodyFormat) {
+      case "text/plain":
+        return (
+          <AnnotationItemPlainText
+            value={annotationBodyValue}
+            handleClick={handleClick}
+          />
+        );
+      case "text/html":
+        return (
+          <AnnotationItemHTML
+            value={annotationBodyValue}
+            handleClick={handleClick}
+          />
+        );
+      case "text/vtt":
+        return (
+          <AnnotationItemVTT
+            label={annotationBody[0].label}
+            vttUri={annotationBody[0].id || ""}
+          />
+        );
+      default:
```

Comment:
@wykhuh Glad you mentioned this, as the `renderItemBody` function could be cleaned up a bit.  I'm not the biggest fan of using a `switch` statements, but seems a decent enough use case for it here.  Since the `switch` is on a type `string`, your regex matcher above could return `undefined`, so not entirely sure it's valid, but TypeScript is not complaining and it does logically read better not to have the image regex check tucked under the `default` (this seems wrong too).  I'll plug your code in 😀

Reply:
That being said, I'm going to slightly re-factor this component and simplify the `renderItemBody` function to only do one thing (`switch`).